mongo-fixes/update-dimension-links/main.go

```diff
@@ -0,0 +1,189 @@
+	InstanceID        string                      `bson:"id,omitempty"                          json:"id,omitempty"`
+	LastUpdated       time.Time                   `bson:"last_updated,omitempty"                json:"last_updated,omitempty"`
+	LatestChanges     *[]models.LatestChange      `bson:"latest_changes,omitempty"              json:"latest_changes,omitempty"`
+	Links             *models.InstanceLinks       `bson:"links,omitempty"                       json:"links,omitempty"`
+	ReleaseDate       string                      `bson:"release_date,omitempty"                json:"release_date,omitempty"`
+	State             string                      `bson:"state,omitempty"                       json:"state,omitempty"`
+	Temporal          *[]models.TemporalFrequency `bson:"temporal,omitempty"                    json:"temporal,omitempty"`
+	TotalObservations *int                        `bson:"total_observations,omitempty"          json:"total_observations,omitempty"`
+	UniqueTimestamp   bson.MongoTimestamp         `bson:"unique_timestamp"                      json:"-"`
+	Version           int                         `bson:"version,omitempty"                     json:"version,omitempty"`
+}
+
+func main() {
+	flag.StringVar(&mongoURL, "mongo-url", mongoURL, "mongoDB URL")
+	flag.Parse()
+
+	ctx := context.Background()
+
+	if mongoURL == "" {
+		log.Event(ctx, "missing mongo-url flag", log.ERROR)
+		return
+	}
+
+	session, err := mgo.Dial(mongoURL)
+	if err != nil {
+		log.Event(ctx, "unable to create mongo session", log.Error(err), log.ERROR)
+		return
+	}
+	defer session.Close()
+
+	session.SetBatch(10000)
+	session.SetPrefetch(0.25)
+
+	log.Event(ctx, "successfully connected to mongo", log.INFO)
+
+	// Get all instances IDs
+	instanceIDs, err := getInstanceIDs(ctx, session)
+	if err != nil {
+		log.Event(ctx, "failed to get all instances", log.Error(err), log.ERROR)
+		return
+	}
+	log.Event(ctx, "successfully retrieved all instance ids", log.INFO)
+
+	// Create a backup collection
+	// dateTime formatted in YYMMDD_HHMMSS
+	dateTime := time.Now().Format("20060102_150405")
+	backupProgressBar := progressbar.Default(int64(len(instanceIDs)), "backup instance")
+	for _, id := range instanceIDs {
+		if err = addInstanceToBackup(ctx, session, id.ID, dateTime); err != nil {
+			log.Event(ctx, "failed to backup instances", log.Error(err), log.ERROR)
+			return
+		}
+		backupProgressBar.Add(1)
+		time.Sleep(40 * time.Millisecond)
+	}
+
+	log.Event(ctx, "successfully backed up all instance documents", log.INFO)
+
+	errorCount := 0
+	updateProgressBar := progressbar.Default(int64(len(instanceIDs)), "updating instance")
+
+	// loop over instances
+	for _, id := range instanceIDs {
+
+		// update the whole instance document
+		if err := updateInstance(ctx, session, id.ID); err != nil {
+			log.Event(ctx, "failed to update dimension of instance", log.Error(err), log.Data{"current_instance_id": id}, log.ERROR)
+			errorCount++
+		}
+
```

Comment:
tempting to have a limit on accepted error numbers
```suggestion
	if errorCount > 10 {
		log.Event(ctx, "too many errors updating instances", log.Error(err), log.ERROR)
		return
	}
```